pandas/_libs/tslib.pyx

```diff
@@ -1577,7 +1614,14 @@ cpdef convert_str_to_tsobject(object ts, object tz, object unit,
                     return obj
                 else:
                     # Keep the converter same as PyDateTime's
-                    ts = Timestamp(obj.value, tz=obj.tzinfo)
+                    obj = convert_to_tsobject(obj.value, obj.tzinfo,
```

Comment:
ok so this is getting cumbersome. why don't you make a separate routine to do these 3 (and I guess go back to consrructing datetime). But why does the 

``ts = Timestamp(obj.value, tz=obj.tzinfo)`` not work here?

Reply:
> why don't you make a separate routine to do these 3 (and I guess go back to consrructing datetime)

OK

> But why does the `ts = Timestamp(obj.value, tz=obj.tzinfo)` not work here?

It works, but the goal is to disentangle `Timestamp` from `_TSObject` logic.  Or more specifically, to make the dependency go in one direction.